pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -180,6 +182,28 @@ func (c *TargetConfigController) getSubstitutionReplacer(operatorSpec *operatorv
 	), nil
 }
 
+func (c *TargetConfigController) checkDependencies() error {
+	// Ensure kube-controller-manager has rotated csr-controller-ca before rolling out.
+	// Waiting here is very important in single node as we want to avoid disruption until
+	// key resources have rotated.
+	csrControllerCAConfigMap, err := c.configMapLister.ConfigMaps(operatorclient.GlobalMachineSpecifiedConfigNamespace).Get("csr-controller-ca")
+	if err != nil {
+		return err
+	}
+
+	csrControllerCAConfigMapCopy := csrControllerCAConfigMap.DeepCopy()
```

Comment:
Maybe skip the copy if no modification is performed?

Reply:
was just trying to be extra cautious with the object but point taken.